test_conformance/profiling/enqueueSyncFunctions.cpp

```diff
@@ -0,0 +1,286 @@
+    test_error(error, "Profiling sync2 QUEUED failed");
+    error = clGetEventProfilingInfo(sync_event2, CL_PROFILING_COMMAND_SUBMIT,
+                                    sizeof(t2Submit), &t2Submit, nullptr);
+    test_error(error, "Profiling sync2 SUBMIT failed");
+    error = clGetEventProfilingInfo(sync_event2, CL_PROFILING_COMMAND_START,
+                                    sizeof(t2Start), &t2Start, nullptr);
+    test_error(error, "Profiling sync2 START failed");
+    error = clGetEventProfilingInfo(sync_event2, CL_PROFILING_COMMAND_END,
+                                    sizeof(t2End), &t2End, nullptr);
+    test_error(error, "Profiling sync2 END failed");
+    if (check_times(t2Queued, t2Submit, t2Start, t2End, device) != TEST_PASS)
+    {
+        log_error("Timestamp monotonicity failed for sync event 2\n");
+        test_status = TEST_FAIL;
+    }
+
+    return test_status;
+}
+
+int test_enqueue_function(cl_device_id device, cl_context context,
+                          int (*fn)(cl_command_queue command_queue,
+                                    cl_uint num_events_in_wait_list,
+                                    const cl_event *event_wait_list,
+                                    cl_event *event))
+{
+    cl_int error;
+
+    cl_command_queue_properties dev_props = 0;
+    error = clGetDeviceInfo(device, CL_DEVICE_QUEUE_PROPERTIES,
+                            sizeof(dev_props), &dev_props, nullptr);
+    test_error(error, "clGetDeviceInfo(CL_DEVICE_QUEUE_PROPERTIES) failed");
+    bool ooo_supported =
+        (dev_props & CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE) != 0;
+
+    if (!ooo_supported)
+        log_info("Device does NOT support out-of-order queues: OOO phase "
+                 "skipped.\n");
+    else
+        log_info(
+            "Device supports out-of-order queues: running both variants.\n");
+
+    clCommandQueueWrapper in_order_queue = clCreateCommandQueue(
+        context, device, CL_QUEUE_PROFILING_ENABLE, &error);
+    test_error(error, "Failed to create in-order profiling queue");
+
+    int status_in_order =
+        run_enqueue_variant(device, context, in_order_queue, false, fn);
+
+    int status_ooo = TEST_PASS;
+    if (ooo_supported)
+    {
+        clCommandQueueWrapper ooo_queue = clCreateCommandQueue(
+            context, device,
+            CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | CL_QUEUE_PROFILING_ENABLE,
+            &error);
+        test_error(error, "Failed to create out-of-order profiling queue");
+        status_ooo = run_enqueue_variant(device, context, ooo_queue, true, fn);
+    }
+
+    return (status_in_order == TEST_FAIL || status_ooo == TEST_FAIL)
+        ? TEST_FAIL
+        : TEST_PASS;
+}
+
+REGISTER_TEST(enqueue_marker)
+{
+    int (*foo)(cl_command_queue command_queue, cl_uint num_events_in_wait_list,
+               const cl_event *event_wait_list, cl_event *event);
+    foo = clEnqueueMarkerWithWaitList;
+    return test_enqueue_function(device, context, foo);
```

Comment:
Is there a need for `foo`, or can this be simplified?

```suggestion
    return test_enqueue_function(device, context, clEnqueueMarkerWithWaitList);
```